sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreCoordinator.scala

```diff
@@ -168,9 +220,99 @@ private class StateStoreCoordinator(override val rpcEnv: RpcEnv)
         storeIdsToRemove.mkString(", "))
       context.reply(true)
 
+    case SnapshotUploaded(providerId, version, timestamp) =>
+      stateStoreSnapshotVersions.put(providerId, SnapshotUploadEvent(version, timestamp))
```

Comment:
I think we should only update if it is a higher version than the current recorded version. To avoid a situation where provider was moved to another executor but previous executor maintenance thread uploaded a old version after new executor uploaded new version. Chances of this happening is low but not zero, so lets handle and add comment.

Reply:
Gotcha, yeah better to stay on the safe side for this 👍